pkg/operator/controller/ingress/deployment.go

```diff
@@ -282,6 +282,11 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		// node.  Thus no affinity policy is required when using
 		// HostNetwork.
 	case operatorv1.PrivateStrategyType, operatorv1.LoadBalancerServiceStrategyType, operatorv1.NodePortServiceStrategyType:
+		// non-HA ingress controllers should have default deployment strategy with no affinity policy
+		if singleReplica(ingressConfig, infraConfig) {
```

Comment:
@Miciah actually `if desiredReplicas == 1` should work for our use case (SingleReplica infrastructure, one worker node where ingress controller runs, and "NodePortService" endpoint publishing strategy). We did not try to consider the case of SingleReplica topology with one ingress controller desired replica and more than one node available for it to run on. If the proposed solution really ensures `rolling updates schedule the new pod on the same node as the old pod` and you consider this as the desired behavior in multi-node non-HA clusters, then we can leave the code fix as-is. Otherwise we do not mind switching to `if desiredReplicas == 1`.

Reply:
I mean, using `if singleReplica(ingressConfig, infraConfig)` and not `if desiredReplicas == 1` seems to me to be the right thing to do, but it isn't necessarily obvious why it's the right thing, and the casual reader might perceive `singleReplica` to be unnecessary complexity.  It would be helpful to add a comment like the following:
```go
		// We need to check the cluster topology in order to determine
		// whether the affinity rule is required.  Checking
		// desiredReplicas would be incorrect because we need the rule
		// on multi-node clusters, even if desiredReplicas is 1, so that
		// a rolling update schedules the new pod to the same node as
		// the old pod.
```

Reply:
now it makes total sense, will add the comment, thanks!